atlasdb-api/src/main/java/com/palantir/atlasdb/spi/KeyValueServiceConfig.java

```diff
@@ -39,9 +38,8 @@ public interface KeyValueServiceConfig {
      * you've acknowledged the risks and side effects mentioned in the relevant NamespaceDeleter docs (e.g.,
      * CassandraNamespaceDeleter)
      */
-    @Value.Default
-    @JsonProperty("enableNamespaceDeletionDangerousIKnowWhatIAmDoing")
-    default boolean enableNamespaceDeletion() {
+    @Value.Default // Without this annotation, implementors cannot add this property as part of an immutable builder.
```

Comment:
this is definitely really weird to have on a non-immutable - definitely in favour of having each downstream implement it

As for the name - I think this is the better approach; depending upon others to add the json annotation seems errorprone.

Reply:
Yep, figured. In that case, I won't set a default and take the couple breaks (internal migration and proxy test config classes) and fix them myself when the dep autobumps. If I set a default here without the annotation, that can lead to the subtle bug of not deserialising the new field correctly because it's not in the immutable (Tested via CKVSCTest)